doc/explanation/api/reactivity.md

```diff
@@ -0,0 +1,177 @@
+
+1. They are **data-driven**, i.e. the UI updates automatically in response to changes in the data model. In Excel the data model is represented by value of a cell, while React has use state hooks. In Panel the data model is defined by Parameters, i.e. components and user classes define some parameters which drive changes in the UI. In all these frameworks this is achieved by what we will call **data binding**, how precisely this works will be the primary focus of this section.
+
+2. They are **declarative**, i.e. the user, defines what should be rendered and the reactive framework figures out how to efficiently make the required updates. In Excel this is simple, it simply has to re-evaluate any formulas that depend on the changed inputs and then update the cell's output. In React this can be significantly more involved, requiring diffing of the document model and then figuring out the most efficient updates to reflect the latest changes. In Panel this works either by defining a function who's output is diffed or by binding references to a declarative component (if you don't know what that means yet, don't worry, that's what this guide is for).
+
+The alternative to the reactive approach is an event-driven approach, involving callbacks that are triggered by user interactions or other events. This approach usually relies on **imperative** rather than **declarative** programming, i.e. you define step-by-step what should happen when a particular action is triggered. This is usually significantly more cumbersome, especially as an application grows in size and complexity. Even in a relatively simple application where we gather user inputs from multiple widgets and then perform some update we quickly run into problems. Let us take a simple example of an application that has three widgets that gather user input. In a callback based approach we have to register a callback with each of the three widgets and then perform some action as a result:
+
+![Diagram of callback based UI code](../../_static/images/callbacks_diagram.png)
+
+The problem we immediately run into if some action we want to take requires more than one of the inputs is how we coordinate these changes. This is why many older callback based frameworks rely on what is called the Model-View-Controller (MVC) pattern, where the **view** that the user interacts with, gathers the inputs, which are fed to a **controller** that is able to coordinate the various events and manipulate the underlying model. This requires significant state handling, which in practice is only really feasible by implement a class based controller.
+
+In the reactive approach things are much simpler, the inputs are bound to a declarative function or component that immediately reflects the required changes in the rendered view:
+
+![Diagram of reactive UI code](../../_static/images/reactive_diagram.png)
```

Comment:
I like the figures. They explain well that the *reactive*/ *declarative* approach is simpler to reason about.

I would add big fat figure titles to make it even clearer which approach the figure refers to.

Reply:
Would it be possible to have two tabs. On the first tab this image. On the second tab e2e code that can run illustrating the picture?